### DISCUSSION TITLE
Retention period of toots from other instances (GDPR)

### DISCUSSION THREAD

**Discussion by rriemann** [mastodon/mastodon#18763] [Q&A]
Dear all,

The federated timeline and the tag search show toots from local and federated users. How far back in history does this go? I try to find out the retention periods applicable to the local instance to add details to data protection notices for GDPR compliance.

Is the search only using local data or will it while scrolling back again contact other instances?

Best,   
Robert

**Comment by trwnh** [comment]
local data only. the public timeline contains statuses that your instance is aware of. it does not contact other instances.

**Reply by rriemann** [reply]
And the toot data from other instances is stored locally forever?

**Reply by trwnh** [reply]
i believe it is stored until the admin runs a cleanup task to purge old statuses. otherwise, yes, the statuses remain in the database unless a Delete is received.

**Comment by nightpool** [comment]
Mastodon operates on a federated model very similar to email. When a user makes a post, they send the JSON representation of that post to all of their followers. Their followers store that post, optionally indexing it for search, bookmarking, etc. While Mastodon has certain protections around indexing, visibility, etc. for posts it receives, fundamentally once you send a post to one of your followers it's out of your hands, and it's up to the policies of the remote instance to decide what to do with it. Although remote instances have certain technically-agreed-upon obligations to e.g. not disclose private posts to other users or to delete them when requested by the original poster, it's still up to that remote instance to carry out those functions appropriately, since there's no technical way to enforce that compliance.

**Reply by rriemann** [reply]
I understand that the ActivityPub protocol proposes APIs for the dissemination of content and that the retention is a decision taken by the implementing software. So where can I find how the software Mastodon deals with deletion of old remote posts from the local database?

I read https://docs.joinmastodon.org/admin/tootctl/ and  could not identify a command that would allow to remove remote posts older than a certain age.

**Reply by Gargron** [reply]
We have not implemented such a command yet.

**Comment by rriemann** [comment]
https://github.com/mastodon/mastodon/pull/19232 should fix this. Thank you!
